libdevice/sanitizer/tsan_rtl.cpp

```diff
@@ -0,0 +1,337 @@
+//==--- tsan_rtl.cpp - device thread sanitizer runtime library -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "include/tsan_rtl.hpp"
+#include "atomic.hpp"
+#include "device.h"
+#include "spirv_vars.h"
+
+DeviceGlobal<void *> __TsanLaunchInfo;
+
+#if defined(__SPIR__) || defined(__SPIRV__)
+
+static const __SYCL_CONSTANT__ char __tsan_print_generic_to[] =
+    "[kernel] %p(4) - %p(%d)\n";
+
+static const __SYCL_CONSTANT__ char __tsan_print_raw_shadow[] =
+    "[kernel] %p(%d) -> %p: {%x, %x, %x, %x}\n";
+
+static const __SYCL_CONSTANT__ char __tsan_print_shadow_value[] =
+    "[kernel] %p(%d) : {size: %d, access: %x, sid: %d, clock: %d, is_write: "
+    "%d}\n";
+
+static const __SYCL_CONSTANT__ char __tsan_print_unsupport_device_type[] =
+    "[kernel] Unsupport device type: %d\n";
+
+static const __SYCL_CONSTANT__ char __tsan_report_race[] =
+    "[kernel] data race (%s:%d) in %s\n";
+
+#define TsanLaunchInfo                                                         \
```

Comment:
Could you move the TsanLaunchInfo macro right after __TsanLaunchInfo def?

Reply:
done